### ISSUE TITLE
[css-text-decor] text-underline-position auto in vertical text

### ISSUE THREAD

**Issue by r12a** [w3c/csswg-drafts#1198]
2.6. Text Underline Position: the text-underline-position property > Auto
https://drafts.csswg.org/css-text-decor-3/#underline-auto

If you apply text-underline-position: auto, should the underline appear on the right of Japanese (and Mongolian) text, as is recommended for the default text-decoration(-line):underline values?

If so, it would be good to make that clear (and btw how 'alphabetic baseline' relates to cjk positioning of an underline in vertical text).

**Comment by fantasai** [comment]
No, `auto` does not switch sides. It only allows the position to be automatically adjusted on that side

**Comment by r12a** [comment]
Could we add a note or some text to clarify that?

**Comment by fantasai** [comment]
See [testcase](https://software.hixie.ch/utilities/js/live-dom-viewer/?%3C!DOCTYPE%20html%3E%0A%3Chtml%20style%3D%22writing-mode%3A%20vertical-rl%22%3E%0A%3Cstyle%3Ep%20%7B%20text-decoration%3A%20underline%3B%20text-underline-position%3A%20auto%3B%20%7D%3C%2Fstyle%3E%0A%3Cp%3EUntagged%3C%2Fp%3E%0A%3Cp%20lang%3Dja%3EJapanese%3C%2Fp%3E%0A%3Cp%20lang%3Dzh%3EChinese%3C%2Fp%3E%0A%3Cp%20lang%3Dko%3EKorean%3C%2Fp%3E%0A).

This is currently not allowed per spec; instead the spec has UA rules that set the behavior per language.

Do we want to

- Update the definition of `auto` to match Firefox? 
- Or update implementations to match the spec?

**Comment by hober** [comment]
Firefox (and Chrome)'s current behavior is better than the specced behavior, so I think we should update the definition of `auto`.

**Comment by nt1m** [comment]
I think the `auto` value should stay as is, and we should instead rely on the UA stylesheet to do the right thing: https://drafts.csswg.org/css-text-decor-3/#default-stylesheet

This is what WebKit implements, and I think it's sensible. 

My suggestion would be to resolve with no change.

**Comment by nt1m** [comment]
Whatever ends up being done here should be consistent with text-emphasis-position

**Comment by jfkthame** [comment]
> Firefox (and Chrome)'s current behavior is better than the specced behavior, so I think we should update the definition of `auto`. Given their current behavior, it's presumably web-compatible to make such a spec change.

FWIW, it looks like Firefox, at least, has been doing this for 9 years (since https://bugzilla.mozilla.org/show_bug.cgi?id=1175789, long before `text-underline-position` was supported), and as far as I recall we haven't seen any issues with this being the default (now `auto`) behavior.

**Comment by dletorey** [comment]
I have a question about `text-emphasis-position: auto;` I assume that `auto` is the initial value, is my assumption here correct?

**Comment by jfkthame** [comment]
> I have a question about `text-emphasis-position: auto;` I assume that `auto` is the initial value, is my assumption here correct?

I believe that was the intention, yes.
